client/app/bundles/course/assessment/submission/containers/QuestionGrade.tsx

```diff
@@ -0,0 +1,181 @@
+
+  if (!grading) return null;
+
+  const dirty = (grading.originalGrade ?? 0) !== (grading.grade ?? 0);
+
+  const handleUpdateGrade = (
+    id: number,
+    grade: number | string | null,
+  ): void => {
+    dispatch(updateGrade(id, grade, bonusAwarded));
+  };
+
+  const processValue = (value: string, drafting: boolean = false): void => {
+    if (value.trim() === '') {
+      return handleUpdateGrade(questionId, null);
+    }
+
+    if (drafting && !isValidDecimal(value)) {
+      return undefined;
+    }
+
+    const parsedValue = parseFloat(value);
+
+    if (!drafting && (Number.isNaN(parsedValue) || parsedValue < 0)) {
+      return handleUpdateGrade(questionId, null);
+    }
+
+    if (parsedValue >= maxGrade) {
+      return handleUpdateGrade(questionId, maxGrade);
+    }
+
+    return handleUpdateGrade(questionId, drafting ? value : parsedValue);
+  };
+
+  const stepGrade = (delta: number): void => {
+    const parsedValue =
+      typeof grading.grade === 'string'
+        ? parseFloat(grading.grade)
+        : grading.grade ?? 0;
+    const newGrade = Math.max(Math.min(parsedValue + delta, maxGrade), 0);
+    return handleUpdateGrade(questionId, newGrade);
+  };
+
+  const renderQuestionGradeField = (): JSX.Element => (
+    <div className="flex w-full items-center space-x-2">
+      <div className="flex items-center space-x-4">
+        <TextField
+          className="w-40"
+          hiddenLabel
+          inputProps={{ className: 'grade' }}
+          onBlur={(e): void => processValue(e.target.value)}
+          onChange={(e): void => {
+            processValue(e.target.value, true);
+            debouncedSaveGrade(questionId);
+          }}
+          onKeyDown={(e): void => {
+            if (e.key === 'ArrowUp') {
+              e.preventDefault();
+              stepGrade(GRADE_STEP);
+            }
+            if (e.key === 'ArrowDown') {
+              e.preventDefault();
+              stepGrade(-GRADE_STEP);
+            }
+            debouncedSaveGrade(questionId);
+          }}
+          placeholder={
+            typeof grading.originalGrade === 'number'
+              ? grading.originalGrade.toString()
+              : grading.originalGrade
+          }
+          size="small"
+          value={grading.grade ?? ''}
+          variant="filled"
+        />
+
+        <Typography color="text.disabled" variant="body2">
+          /
+        </Typography>
+
+        <Typography variant="body2">{maxGrade}</Typography>
+      </div>
+
+      <div className="px-4 space-x-4">
+        {grading.prefilled && (
+          <Tooltip title={intl.formatMessage(translations.gradePrefilledHint)}>
+            <Chip
+              className="slot-1-neutral-400 border-slot-1 text-slot-1"
+              label={intl.formatMessage(translations.gradePrefilled)}
+              size="small"
+              variant="outlined"
+            />
+          </Tooltip>
+        )}
+
+        {dirty ? (
```

Comment:
Show that a grade is being saved as well

Reply:
I'm sorry but seems like I don't really understand your concern here. If the grade has been saved, it's indeed shown as "Saved", as follows
<img width="1512" alt="Screenshot 2023-10-03 at 11 06 46 PM" src="https://github.com/Coursemology/coursemology2/assets/16359075/6cef9e81-6292-48a4-b9ed-9800b611e2bc">